src/lib_json/json_writer.cpp

```diff
@@ -165,7 +166,7 @@ JSONCPP_STRING valueToString(double value, bool useSpecialFloats, unsigned int p
       len = snprintf(buffer, sizeof(buffer), useSpecialFloats ? "Infinity" : "1e+9999");
     }
   }
-  assert(len >= 0);
+  assert(len >= 0 && len < sizeof(buffer));
```

Comment:
This assert happens way too late. We already used `len` as a range limit above in the calls to fixNumericLocale etc.
Also the "%%.%ug" 2-stage snprintf is wasteful. We can just use .* precision.

I would like to refactor this function before changing it. I'll send a PR.

Reply:
Thanks Billy, I will waiting for your code refactor.